lib/themes/dosomething/paraneue_dosomething/js/utilities/ApiClient.js

```diff
@@ -137,29 +106,45 @@ class ApiClient {
   post(path, body = {}) {
     const url = `${this.url}${path}`;
 
-    // @TODO: little experiment with console output for providing more information when
-    // API events are happening. console.log() statement are stripped on Prod builds.
-    // Might be useful to have a set of defined styles on a JS object that can be passed
-    // through: warning, notice, succes, etc...
-    console.log('%c ApiClient POST Request Body: \n' + JSON.stringify(body),
-      'background-color: rgba(105,157,215,0.5); color: rgba(33,70,112,1); line-height: 1.5;'
-    );
-
-    return window.fetch(url, {
-      method: 'POST',
-      headers: this.headers,
-      credentials: 'same-origin',
+    return this.send('POST', url, {
       body: JSON.stringify(body)
-    })
-      .then(this.checkStatus)
-      .then(this.parseJson)
-      .then(function(response) {
-        return response;
     });
   }
 
-  send() {
-    // @TODO: setup this send method that all other request methods call.
+  /**
+   * Send an API request using fetch() and return response.
+   *
+   * @param  {String} method
+   * @param  {String} url
+   * @param  {Object} options
+   * @return {Promise}
+   */
+  send(method, url, options = {}) {
+    const defaults = {
+      method: method,
+      headers: this.headers,
+      credentials: 'same-origin'
+    };
+
+    if (! isEmpty(options)) {
+      Object.assign(defaults, options);
+    }
+
+    // Developer Output:
+    console.log('%c ApiClient %s Request:',
+      'background-color: rgba(105,157,215,0.5); color: rgba(33,70,112,1); display: block; font-weight: bold; line-height: 1.5;',
+      method
+    );
+    console.log('URL: \n%s', url);
+    console.log('Options: \n%o', defaults);
+
+    return window.fetch(url, defaults)
+      .then(this.checkStatus)
+      .then(this.parseJson)
+      .catch((data) => {
+        console.log(data);
+        console.log('um, there was an error! handle that shizzzzz!');
```

Comment:
ah, jes. I need to still flesh out the error handling stuff, but totally makes sense to use that instead!